specification/compute/resource-manager/Microsoft.Compute/stable/2020-06-30/disk.json

```diff
@@ -0,0 +1,2754 @@
+                "value": "FromImage",
+                "description": "Create a new disk from a platform image specified by the given imageReference or galleryImageReference."
+              },
+              {
+                "value": "Import",
+                "description": "Create a disk by importing from a blob specified by a sourceUri in a storage account specified by storageAccountId."
+              },
+              {
+                "value": "Copy",
+                "description": "Create a new disk or snapshot by copying from a disk or snapshot specified by the given sourceResourceId."
+              },
+              {
+                "value": "Restore",
+                "description": "Create a new disk by copying from a backup recovery point."
+              },
+              {
+                "value": "Upload",
+                "description": "Create a new disk by obtaining a write token and using it to directly upload the contents of the disk."
+              }
+            ]
+          },
+          "description": "This enumerates the possible sources of a disk's creation."
+        },
+        "storageAccountId": {
+          "type": "string",
+          "description": "Required if createOption is Import. The Azure Resource Manager identifier of the storage account containing the blob to import as a disk."
+        },
+        "imageReference": {
+          "$ref": "#/definitions/ImageDiskReference",
+          "description": "Disk source information."
+        },
+        "galleryImageReference": {
+          "$ref": "#/definitions/ImageDiskReference",
+          "description": "Required if creating from a Gallery Image. The id of the ImageDiskReference will be the ARM id of the shared galley image version from which to create a disk."
+        },
+        "sourceUri": {
+          "type": "string",
+          "description": "If createOption is Import, this is the URI of a blob to be imported into a managed disk."
+        },
+        "sourceResourceId": {
+          "type": "string",
+          "description": "If createOption is Copy, this is the ARM id of the source snapshot or disk."
+        },
+        "sourceUniqueId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "If this field is set, this is the unique id identifying the source of this resource."
+        },
+        "uploadSizeBytes": {
+          "type": "integer",
+          "format": "int64",
+          "description": "If createOption is Upload, this is the size of the contents of the upload including the VHD footer. This value should be between 20972032 (20 MiB + 512 bytes for the VHD footer) and 35183298347520 bytes (32 TiB + 512 bytes for the VHD footer)."
+        },
+        "logicalSectorSize": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Support customers creating ultra disks with logicalSectorSize 512E. Supported values are 512 and 4096 (default value)."
```

Comment:
Logical sector size in bytes for Ultra disks. Supported values are 512 ad 4096. 4096 is the default.